pyerrors/input/sfcf.py

```diff
@@ -29,266 +53,306 @@ def read_sfcf(path, prefix, name, **kwargs):
-    im -- if True, read imaginary instead of real part of the correlation function.
-    b2b -- if True, read a time-dependent boundary-to-boundary correlation function
-    names -- Alternative labeling for replicas/ensembles. Has to have the appropriate length
-    ens_name : str
-        replaces the name of the ensemble
-    """
-
-    if kwargs.get('im'):
-        im = 1
-        part = 'imaginary'
-    else:
-        im = 0
-        part = 'real'
-
-    if kwargs.get('b2b'):
-        b2b = 1
-    else:
-        b2b = 0
-
-    T = 0
-    ls = []
-    for (dirpath, dirnames, filenames) in os.walk(path):
-        ls.extend(dirnames)
-        break
-    if not ls:
-        raise Exception('Error, directory not found')
-    # Exclude folders with different names
-    for exc in ls:
-        if not fnmatch.fnmatch(exc, prefix + '*'):
-            ls = list(set(ls) - set([exc]))
+        if not ls:
+            raise Exception('Error, directory not found')
+        # Exclude folders with different names
+        for exc in ls:
+            if not fnmatch.fnmatch(exc, prefix + '*'):
+                ls = list(set(ls) - set([exc]))
     if len(ls) > 1:
         ls.sort(key=lambda x: int(re.findall(r'\d+', x[len(prefix):])[0]))  # New version, to cope with ids, etc.
-    replica = len(ls)
+    if not appended:
+        replica = len(ls)
+    else:
+        replica = len([l.split(".")[-1] for l in ls])//len(set([l.split(".")[-1] for l in ls]))
+    print('Read', part, 'part of', name, 'from', prefix[:-1], ',', replica, 'replica')
     if 'names' in kwargs:
         new_names = kwargs.get('names')
+        if len(new_names)!=len(set(new_names)):
+            raise Exception("names are not unique!")
         if len(new_names) != replica:
             raise Exception('Names does not have the required length', replica)
     else:
         # Adjust replica names to new bookmarking system
+
         new_names = []
-        for entry in ls:
-            idx = entry.index('r')
-            if 'ens_name' in kwargs:
-                new_names.append(kwargs.get('ens_name') + '|' + entry[idx:])
+        if not appended:
+            for entry in ls:
+                try:
+                    idx = entry.index('r')
+                except:
+                    raise Exception("Automatic recognition of replicum failed, please enter the key word 'names'.")
+                    
+                if 'ens_name' in kwargs:
+                    new_names.append(kwargs.get('ens_name') + '|' + entry[idx:])
+                else:
+                    new_names.append(entry[:idx] + '|' + entry[idx:])
+        else:
+            
+            for exc in ls:
+                if not fnmatch.fnmatch(exc, prefix + '*.'+name):
+                    ls = list(set(ls) - set([exc]))
+            ls.sort(key=lambda x: int(re.findall(r'\d+', x)[-1]))
+            for entry in ls:
+                myentry = entry.removesuffix("."+name)
```

Comment:
This is a feature which was only added in python 3.9. As we support all version >=3.6 this would have to be changed. See for example https://stackoverflow.com/a/1038845